cpp/src/join/conditional_join_kernels.cuh

```diff
@@ -220,8 +220,8 @@ __global__ void conditional_join(table_device_view left_table,
                                                          join_output_r);
         __syncwarp(activemask);
         if (0 == lane_id) { current_idx_shared[warp_id] = 0; }
-        __syncwarp(activemask);
       }
+      __syncwarp(activemask);
```

Comment:
This `__syncwarp` and the previous one both look to be using the wrong active mask. Not all threads in active mask could have evaluated the `current_idx_shared[warp_id] + detail::warp_size >= output_cache_size`. So we should use ballot_sync to compute the submask for the if statement

Reply:
You are right of course. Did I learn nothing from the last bug? :-)